config/routechangerconfig_test.go

```diff
@@ -71,3 +99,22 @@ func Test_routeChangerConfigSeveralItems(t *testing.T) {
 		})
 	}
 }
+
+func TestRouteChangerYaml(t *testing.T) {
+	m := routeChangerConfig{}
+	err := yaml.Unmarshal([]byte(`"#Source[(](.*)[)]#ClientIP($1)#"`), &m)
+	if err != nil {
+		t.Errorf("Failed to yaml unmarshal: %v", err)
+	}
+	if len(m) != 1 {
+		t.Errorf("Failed to get correct routechanger config: %d != 1", len(m))
+	}
+}
```

Comment:
I think I've used (misused?) `Test_NewConfig` to validate yaml parsing https://github.com/zalando/skipper/pull/1933

Reply:
I think the test is fine, but testing different cases to have more/all call paths covered requires sometimes more special tests.